### PR TITLE
RE2022-268: Add basic provenance information when saving sets

### PR DESCRIPTION
**Pull request by MrCreosote**
... as per /design/provenance.md

Also throw an error if the description is > 800 chars

Match information is still do do

### CODE REVIEW

File: src/service/processing_selections.py [hunk 1]
```diff
@@ -244,7 +248,8 @@ async def save_set_to_workspace(
     workspace_id: int,
     object_name: str,
     ws_type: str,
-    description: str = None
+    service_ver: str = None,
+    description: str = None,
```
Comment (Xiangs18) [inline src/service/processing_selections.py:252]: Need to update doc string as well.

File: src/service/processing_selections.py [hunk 2]
```diff
@@ -21,6 +21,10 @@
 
 MAX_SELECTION_IDS = 10000
 
+# The set descriptions go in the workspace metadata, which has a maximum size of 900 for the
```
Comment (Xiangs18) [inline src/service/processing_selections.py:24]: 800?

File: src/service/processing_selections.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/processing_selections.py:252]: I've got it done locally; it'll be in the next PR. Cool?

File: src/service/processing_selections.py (hunk 2 above)
Comment (MrCreosote) [reply src/service/processing_selections.py:24]: Workspace limit is 900 for len(key) + len(value), so max description is 900 - len(key). I just rounded down to 800

**Review by Xiangs18 — APPROVED** [review]
LGTM!

**Pull request merged**